homeassistant/components/minecraft_server/__init__.py

```diff
@@ -23,8 +30,19 @@
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up Minecraft Server from a config entry."""
 
+    # Check and create API instance.
+    try:
+        api = MinecraftServer(
+            entry.data.get(CONF_TYPE, MinecraftServerType.JAVA_EDITION),
+            entry.data[CONF_ADDRESS],
+        )
```

Comment:
Please run the creation of the server in the executor as the lookup is performed too